src/main/java/org/fogbowcloud/saps/engine/core/model/ImageTask.java

```diff
@@ -66,6 +66,27 @@ public ImageTask(String taskId, String dataset, String region, Date imageDate,
 		this.error = error;
 	}
 
+	public ImageTask(JSONObject imageTaskJsonObject) throws JSONException {
```

Comment:
or, forward the call to this(imageTaskJsonObject.getString("taskId"), imageTaskJsonObject.getString("dataset") ..)

Reply:
Agree